File: OCH_Tao/ABC/B/234.py
from itertools import combinations
from math import sqrt
N = int(input())
L = []
for i in range(N):
  L.append(list(map(int,input().split())))
C = combinations(L,2)
ans = 0
for i in C:
  ans=max(ans,sqrt((i[0][0]-i[1][0])**2+(i[0][1]-i[1][1])**2))
print(ans)

File: OCH_Tao/ABC/B/234.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = parseInt(input.shift(), 10);
let L = [];
for (const i of input) {
  L.push(i.split(" ").map(x => parseInt(x, 10)));
}
let ans = 0;
for (let i = 0; i < N - 1; i++) {
  for (let j = i + 1; j < N; j++) {
    ans = Math.max(ans, Math.sqrt((L[i][0] - L[j][0]) ** 2 + (L[i][1] - L[j][1]) ** 2));
  }
}
console.log(ans);